realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -823,7 +822,7 @@ public void migrationException_realmListChanged() throws IOException {
             fail();
         } catch (RealmMigrationNeededException ignored) {
             assertThat(ignored.getMessage(),
-                    CoreMatchers.containsString("Invalid RealmList type for field 'cats': 'class_Dog' expected "));
+                    CoreMatchers.containsString("Property 'CatOwner.cats' has been changed from 'array' to 'array'"));
```

Comment:
Seems that it was changed from `RealmList<Cat>` to `RealmList<Dog>`, but the exception doesn't seem to help with this fact.